packages/muze-axis/src/cartesian-axis/helper.js

```diff
@@ -312,15 +312,39 @@ export const calculateContinousSpace = (context) => {
     };
 };
 
+/**
+ * Overwrites domain with user defined domain (if present)
+ * @param {Object} context reference to current axes
+ * @param {Array} domain default domain
+ *
+ * @return {Array} domain
+ */
+export const getValidDomain = (context, domain) => {
+    const { domain: userDom } = context.config();
+
+    return userDom || domain;
+};
+
 export const setContinousAxisDomain = (context, domain) => {
-    const { nice, domain: userDom } = context.config();
-    if (userDom) {
-        domain = userDom;
-    }
-    if (domain.length && domain[0] === domain[1]) {
-        domain = [0, +domain[0] * 2];
+    const { nice } = context.config();
+    const scale = context.scale.bind(context);
+
+    scale().domain(domain);
+    nice && scale().nice();
+    context._domain = scale().domain();
+};
+
+/**
+ * Checks if any of the properties have changed between two objects
+ * @param {Object} obj first object
+ * @param {Object} obj1 second object
+ * @param {Array} properties properties to be compared between two objects
+ *
+ * @return {Boolean} boolean value
+ */
+export const hasAxesConfigChanged = (obj, obj1, properties) => {
+    if (!obj || !obj1 || !Object.keys(obj).length || !Object.keys(obj1).length) {
```

Comment:
This check is erroneous and redundant.
1) Redundant: You can eliminate !obj and !obj1 by providing default values in the function (obj = {}, obj1 = {})
2) Erroneous: It's possible that the object has some keys, but not the property to be checked, and that the value of the property is null:
obj: {
a: 10,
b: null
}

obj1: {
c: 100
}

What happens in such a case? Also, if the second object has no keys, it could mean that object has been reset. Rethink the logic. If the logic right now is as intended, just rewrite the function by reducing if-else checks(they reduce scalability)

Reply:
Since we are checking for equality of some properties between old axis config and new axis config. If one of the objects in missing a field while the other has it, it means the property has been updated.